core/src/main/scala/kafka/server/FetchDataInfo.scala

```diff
@@ -19,4 +19,5 @@ package kafka.server
 
 import kafka.message.MessageSet
 
-case class FetchDataInfo(fetchOffsetMetadata: LogOffsetMetadata, messageSet: MessageSet)
+case class FetchDataInfo(fetchOffsetMetadata: LogOffsetMetadata, messageSet: MessageSet,
+                         messageSetIncomplete: Boolean = false)
```

Comment:
To be clearer, messageSetIncomplete probably should be firstMessageSetIncomplete?